flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/client/HiveShimV310.java

```diff
@@ -233,4 +240,23 @@ public LocalDate toFlinkDate(Object hiveDate) {
 			throw new FlinkHiveException("Failed to convert to Flink date", e);
 		}
 	}
+
+	@Override
+	public Writable hivePrimitiveToWritable(Object value) throws CatalogException {
+		try {
+			return super.hivePrimitiveToWritable(value);
+		} catch (CatalogException e) {
```

Comment:
Any other good way? exception is a little hacky and low performance.

Reply:
Maybe I can use optional. Let me have a try